db.go

```diff
@@ -1,15 +1,18 @@
 package reform
 
 import (
+	"context"
 	"database/sql"
 	"time"
 )
 
 // DBInterface is a subset of *sql.DB used by reform.
 // Can be used together with NewDBFromInterface for easier integration with existing code or for passing test doubles.
+//
+// It may grow and shrink over time to include only needed *sql.DB methods.
 type DBInterface interface {
-	DBTX
```

Comment:
Hm, so you remove `DBTX` and `Begin` from the public interface. How does it look like from the backward-compatibility perspective?

Reply:
I think adding that interface was a mistake in the first place, but now we are stick with it for a while. I expanded a comment and added removed methods back for one more version.